### PR TITLE
[Fix](schema change) Fix schema change fault when add complex type column

### PR DESCRIPTION
**Pull request by Yukang-Lian**
Problem: An error is encountered when executing a schema change on a unique table to add a column with a complex type, such as bitmap, as documented in https://github.com/apache/doris/issues/31365

Reason: The error arises because the schema change logic erroneously applies an aggregation check for new columns across all table types, demanding an explicit aggregation type declaration. However, unique and duplicate tables inherently assume default aggregation types for newly added columns, leading to this misstep.

Solution: The schema change process for introducing new columns needs to distinguish between table types accurately. For unique and duplicate tables, it should automatically assign the appropriate aggregation type, which, for the purpose of smooth integration with subsequent processes, should be set to NONE.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -314,14 +320,16 @@ public void analyze(boolean isOlap) throws AnalysisException {
         }
 
         // disable Bitmap Hll type in keys, values without aggregate function.
-        if (type.isBitmapType() || type.isHllType() || type.isQuantileStateType()) {
-            if (isKey) {
-                throw new AnalysisException("Key column can not set complex type:" + name);
-            }
-            if (aggregateType == null) {
-                throw new AnalysisException("complex type have to use aggregate function: " + name);
+        if (keysType == null || keysType == KeysType.AGG_KEYS) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:323]: if u update logic here, u should update logic in `ColumnDefinition` too

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run feut

**Comment by Yukang-Lian** [comment]
run buildall

**Review by luwei16 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**